content/200-concepts/050-overview/500-under-the-hood.mdx

```diff
@@ -75,22 +77,31 @@ Run the following command to set the environment variable globally (in this exam
 
    <tab>
 
-   ```terminal
-   export PRISMA_QUERY_ENGINE_BINARY=/custom/my-query-engine-unix
-   ```
+```terminal
+export PRISMA_QUERY_ENGINE_BINARY=/custom/my-query-engine-unix
+```
 
    </tab>
 
    <tab>
 
-   ```terminal
-   set PRISMA_QUERY_ENGINE_BINARY=c:\custom\my-query-engine-windows.exe
-   ```
+```terminal
+set PRISMA_QUERY_ENGINE_BINARY=c:\custom\my-query-engine-windows.exe
+```
 
    </tab>
 
 </TabbedContent>
 
+#### Navigating a corperate firewall with <inlinecode>PRISMA_BINARIES_MIRROR</inlinecode>
+
+You might find that your company operates behind a corporate firewall and you are unable to access the Prisma binaries via the default url `https://binaries.prisma.sh`.
+In such situations the `PRISMA_BINARIES_MIRROR` environment variable can be used to bypass the corporate firewall by proxying the request to the Prisma domain and download the binaries.
+
+```terminal
+PRISMA_BINARIES_MIRROR=https://my-whitelisted-url
+```
```

Comment:
This is not my understanding of this feature, even though it could work.

I think proxying can be achieved with https://www.prisma.io/docs/reference/api-reference/environment-variables-reference#proxy-environment-variables

`PRISMA_BINARIES_MIRROR` main goal is to provide a way for people to do the binary hosting themselves. For example, if a company would want to build their own binaries for any reason (custom platform or they want to run a fork of the binaries or for security point of view they want to build it on their hardware). Then they can create for example an AWS S3 bucket and upload their binaries.

Reply:
You _can_ create a "proxy" in front of our domain with your own domain. But that is just one of multiple use cases of this variable.